engine/src/main/java/org/destinationsol/files/HullConfigManager.java

```diff
@@ -53,21 +57,37 @@ public HullConfigManager(ItemManager itemManager, AbilityCommonConfigs abilityCo
         configToNameMap = new HashMap<>();
     }
 
-    private static Vector2 readVector2(JsonValue jsonValue, String name, Vector2 defaultValue) {
-        String string = jsonValue.getString(name, null);
-        return (string == null)
-                ? defaultValue
-                : SolMath.readV2(string);
+    private static @Nullable Vector2 readOptionalVector2(@NotNull JsonValue jsonValue, @Nullable String name, @Nullable Vector2 defaultValue) {
```

Comment:
Not entirely sure about the naming of this method. It makes me feel like it will return `Optional<Vector2>

Reply:
@vampcat If it was to return `Optional`, it would have been named readVector2Optional(). Or that is at least how I feel it,, not quite sure how it corresponds to current standards in DestSol/java in general.